popup.js

```diff
@@ -135,6 +145,66 @@ function minimizeStyles(message) {
 
 /**
  * Removes all style attribute properties that are unneeded for a single
+ *     pseudo element.
+ *
+ * @param {Object} message The message Object that contains the pseudo element
+ *     whose style attributes should be minimized.
+ * @param {Document} doc The Document that contains the rendered HTML.
+ * @param {string} selector The CSS selector for the pseudo element.
+ * @param {Array<string>} finalPseudoElements An array to contain the final
+ *     declaration of pseudo elements.  It will be updated to reflect the pseudo
+ *     element that is being processed.
+ */
+function minimizePseudoElementStyle(
+    message,
+    doc,
+    selector,
+    finalPseudoElements) {
+  var maxNumberOfIterations = 5;
+  var match = selector.match(/^#(.*):(:.*)$/);
+  var id = match[1];
+  var type = match[2];
+  var element = doc.getElementById(id);
+  if (element) {
+    var originalStyleMap = message.pseudoElementSelectorToCSSMap[selector];
+    var requiredStyleMap = {};
+    // We compare the computed style before and after removing the pseudo element
+    // and accumulate the differences in |requiredStyleMap|. Because some
+    // properties affect other properties, such as border-style: solid causing a
+    // change in border-width, we do this iteratively until a fixed-point is
+    // reached (or |maxNumberOfIterations| is hit).
+    for (var i = 0; i < maxNumberOfIterations; i++) {
```

Comment:
Can you add a TODO here to unify this logic with minimizeStyles?
// TODO(sfine): Unify this logic with minimizeStyles.

(it's okay if you never finish the TODO, it's a message to future readers of the code)